### PR TITLE
CDD-1192 adding initial dependencies, config and switching to microse…

### PR DESCRIPTION
**Pull request by stewabel**
…rvice-bootstrap

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import AppDependencies._
+import org.scalastyle.sbt.ScalastylePlugin._
+import sbt.Tests.{Group, SubProcess}
+import uk.gov.hmrc.DefaultBuildSettings.{addTestReportOption, defaultSettings, scalaSettings, targetJvm}
+import scala.util.Properties.envOrElse
+
+name := "customs-wco-declaration"
+
+version := envOrElse("CUSTOMS_WCO_DECLARATION_VERSION", "999-SNAPSHOT")
```
Comment (MihailsC-hmrc) [inline build.sbt]: Please remove `version` as per [Prerequisites](https://confluence.tools.tax.service.gov.uk/display/DTRG/Migrating+To+Coding+In+The+Open#MigratingToCodingInTheOpen-Prerequisites)

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import AppDependencies._
+import org.scalastyle.sbt.ScalastylePlugin._
+import sbt.Tests.{Group, SubProcess}
+import uk.gov.hmrc.DefaultBuildSettings.{addTestReportOption, defaultSettings, scalaSettings, targetJvm}
+import scala.util.Properties.envOrElse
+
+name := "customs-wco-declaration"
```
Comment (MihailsC-hmrc) [inline build.sbt]: Please fix the project name

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+val testDependencies = Seq(hmrcTest, scalaTest, pegDown, scalaTestPlusPlay, wireMock, mockito, customsApiCommonTests)
+
+lazy val AcceptanceTest = config("acceptance") extend Test
+lazy val CdsIntegrationTest = config("it") extend Test
+
+val testConfig = Seq(AcceptanceTest, CdsIntegrationTest, Test)
+
+lazy val testAll = TaskKey[Unit]("test-all")
+lazy val allTest = Seq(testAll := (test in AcceptanceTest)
```
Comment (MihailsC-hmrc) [inline build.sbt]: We haven't yet decided on testing strategy in the open. Is it feasible not to have our custom testing approach in the open?

File: conf/application.conf [hunk 4]
```diff
@@ -1,35 +1,38 @@
...
 # ~~~~~
 
 appName=customs-inventory-linking-imports
 
-# An ApplicationLoader that uses Guice to bootstrap the application.
-play.application.loader = "uk.gov.hmrc.play.bootstrap.ApplicationLoader"
-
-# Primary entry point for all HTTP requests on Play applications
-play.http.requestHandler = "uk.gov.hmrc.play.bootstrap.http.RequestHandler"
+appUrl = "http://"${appName}".protected.mdtp"
```
Comment (cichockimc) [inline conf/application.conf]: can we use it here ?

File: project/plugins.sbt [hunk 5]
```diff
@@ -9,3 +9,15 @@ addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "0.9.0")
...
+
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.5.1")
+
+addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "0.8.0")
+
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "0.10.0")
+
+addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "0.8.0")
+
+addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "0.10.0")
```
Comment (cichockimc) [inline project/plugins.sbt]: new line maybe?

File: project/AppDependencies.scala [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  val pegDown = "org.pegdown" % "pegdown" % pegdownVersion % testScope
+
+  val scalaTestPlusPlay = "org.scalatestplus.play" %% "scalatestplus-play" % scalatestplusVersion % testScope
+
+  val wireMock = "com.github.tomakehurst" % "wiremock" % wireMockVersion % testScope exclude("org.apache.httpcomponents","httpclient") exclude("org.apache.httpcomponents","httpcore")
+
+  val mockito =  "org.mockito" % "mockito-core" % mockitoVersion % testScope
+
+  val customsApiCommon = "uk.gov.hmrc" %% "customs-api-common" % customsApiCommonVersion withSources()
```
Comment (MihailsC-hmrc) [inline project/AppDependencies.scala]: Our library `customs-api-common` hasn't yet been open-sourced. Having it in dependency probably violates the *General* section of [Checklist](https://confluence.tools.tax.service.gov.uk/display/DTRG/Coding+in+the+Open+Reviews#CodingintheOpenReviews-Checklist)

File: conf/app.routes [hunk 7]
```diff
@@ -1,3 +1,3 @@
 # microservice specific routes
 
-GET        /hello-world             @uk.gov.hmrc.customsinventorylinkingimports.controllers.MicroserviceHelloWorld.hello
+
```
Comment (MihailsC-hmrc) [inline conf/app.routes]: If we don't have routes, probably we don't need to have controllers as well.

File: conf/application.conf (hunk 4 above)
Comment (MihailsC-hmrc) [inline conf/application.conf]: Probably we don't need this definition.

File: conf/application.conf [hunk 8]
```diff
@@ -1,35 +1,38 @@
-
-include "backend.conf"
+# Copyright 2017 HM Revenue & Customs
```
Comment (MihailsC-hmrc) [inline conf/application.conf]: Overall I guess configuration needs to be thoroughly reviewed and restructured as per [Migrating To Coding In The Open](https://confluence.tools.tax.service.gov.uk/display/DTRG/Migrating+To+Coding+In+The+Open) and [Checklist](https://confluence.tools.tax.service.gov.uk/display/DTRG/Coding+in+the+Open+Reviews#CodingintheOpenReviews-Checklist)

File: project/AppDependencies.scala [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+  val customsApiCommonTests = "uk.gov.hmrc" %% "customs-api-common" % customsApiCommonVersion % testScope classifier "tests"
```
Comment (MihailsC-hmrc) [inline project/AppDependencies.scala]: I'm not sure if we need common tests in dependency.

File: project/plugins.sbt (hunk 5 above)
Comment (MihailsC-hmrc) [inline project/plugins.sbt]: newline please

File: build.sbt (hunk 3 above)
Comment (stewabel) [reply build.sbt]: Staying with this approach based on this afternoon's discussion

File: conf/application.conf [hunk 10]
```diff
@@ -78,20 +88,16 @@ controllers {
...
   uk.gov.hmrc.customsinventorylinkingimports.controllers.MicroserviceHelloWorld = {
-      needsAuth = false
       needsLogging = false
       needsAuditing = false
     }
-
 }
 
-
 # Evolutions
```
Comment (MihailsC-hmrc) [inline conf/application.conf]: Please delete unwanted `Evolutions` section

File: project/AppDependencies.scala [hunk 11]
```diff
@@ -0,0 +1,32 @@
+import sbt._
+
+object AppDependencies {
+  private val xmlResolverVersion = "1.2"
```
Comment (MihailsC-hmrc) [inline project/AppDependencies.scala]: I would recommend moving definition for customs dependencies below (after) the "standard" HMRC ones.

**Comment by MihailsC-hmrc** [comment]
👍

**Pull request merged**